shell/platform/linux/fl_view.cc

```diff
@@ -241,6 +242,41 @@ static gboolean fl_view_button_release_event(GtkWidget* widget,
   return fl_view_send_pointer_button_event(self, event);
 }
 
+// Implements GtkWidget::scroll_event.
+static gboolean fl_view_scroll_event(GtkWidget* widget, GdkEventScroll* event) {
+  FlView* self = FL_VIEW(widget);
+
+  // TODO(robert-ancell): Update to use GtkEventControllerScroll when we can
+  // depend on GTK 3.24.
+
+  gdouble scroll_delta_x = 0.0, scroll_delta_y = 0.0;
+  if (event->direction == GDK_SCROLL_SMOOTH) {
+    scroll_delta_x = event->delta_x;
+    scroll_delta_y = event->delta_y;
+  } else if (event->direction == GDK_SCROLL_UP) {
+    scroll_delta_y = -1;
+  } else if (event->direction == GDK_SCROLL_DOWN) {
+    scroll_delta_y = 1;
+  } else if (event->direction == GDK_SCROLL_LEFT) {
+    scroll_delta_x = -1;
+  } else if (event->direction == GDK_SCROLL_RIGHT) {
+    scroll_delta_x = 1;
+  }
+
+  // TODO: See if this can be queried from the OS; this value is chosen
+  // arbitrarily to get something that feels reasonable.
+  const int kScrollOffsetMultiplier = 20;
+  scroll_delta_x *= kScrollOffsetMultiplier;
+  scroll_delta_y *= kScrollOffsetMultiplier;
+
+  fl_engine_send_mouse_pointer_event(
+      self->engine, self->button_state != 0 ? kMove : kHover,
```

Comment:
IIRC a pointer signal should always be hover.

Reply:
If you hold down a mouse button and drag and do a scroll during that drag the button state should be `kMove`, right? I copied the GLFW/Windows code here and AFAICT they do the above in `SetEventPhaseFromCursorButtonState`.

Reply:
I thought it was documented to always be hover, but I can't find that, and looking at the framework code it seems that the phase is ignored for signals. I may well be remembering an earlier iteration of the PR where it mattered, as I went through several versions of how to encode signals.